### PR TITLE
Fix: EventBus: Exists prefix not considered for rule matching

### LINKED ISSUE #7305: bug: Bug with EventBus filters
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

I wanted to test an EventBus filter that will trigger a Lambda only when a field doesn't exist or when exists and equals to false.
So I created a Lambda with the following configuration (using Serverless Framework)
This is my Lambda Configuration
```  
test:
    handler: test.handler
    memorySize: 512
    description: test
    events:
      - eventBridge:
          eventBus: ${ssm:/${self:custom.env_name}/infra/event-bus/test, 'default'}
          pattern:
            source:
              - 'other-service'
            detail-type:
              - 'test'
            detail:
              event_name:
                - 'test_event_name'
              test_field:
                - false
       - eventBridge:
            eventBus: ${ssm:/${self:custom.env_name}/infra/event-bus/test, 'default'}
            pattern:
              source:
                - 'other-service'
              detail-type:
                - 'test'
              detail:
                event_name:
                  - 'test_event_name'
                test_field:
                    is_background_job: [{exists: false}]
```

When I push an event, It doesn't trigger the Lambda. 
When I tested it on AWS, it worked perfectly.

### Expected Behavior

To trigger the Lambda

### How are you starting LocalStack?

With a docker-compose file

### Steps To Reproduce

  localstack:
    image: localstack/localstack:1.3.0
    ports:
      - "4566-4583:4566-4583"
      - "${PORT_WEB_UI-4666}:${PORT_WEB_UI-8080}"
      - "8080:8080"
      - "4510:4510"
    environment:
      - LOCALSTACK_API_KEY=${LOCALSTACK_API_KEY- }
      - PORT_WEB_UI=8080
      - PERSISTENCE=0
      - LAMBDA_REMOTE_DOCKER=0
      - LAMBDA_EXECUTOR=docker-reuse
      - LAMBDA_REMOVE_CONTAINERS=true
      - DOCKER_HOST=unix:///var/run/docker.sock
      - LAMBDA_DOCKER_NETWORK=localstack-net
      - KINESIS_ERROR_PROBABILITY=${KINESIS_ERROR_PROBABILITY- }
      - LS_LOG=debug
      - DEBUG=1
    volumes:
      - "/var/run/docker.sock:/var/run/docker.sock"
      - "${LOCALSTACK_VOLUME_DIR:-~/Library/Caches/localstack/volume}:/var/lib/localstack"
    networks:
      - localstack-net

### Environment

```markdown
- OS: ubuntu 20.04
- LocalStack: 1.3.0
```

### Anything else?
I tried to deploy it without the test_field filter:
```
test:
    handler: test.handler
    memorySize: 512
    description: test
    events:
      - eventBridge:
          eventBus: ${ssm:/${self:custom.env_name}/infra/event-bus/test, 'default'}
          pattern:
            source:
              - 'other-service'
            detail-type:
              - 'test'
            detail:
              event_name:
                - 'test_event_name'
 ```
And it worked well.

### LINKED ISSUE #8512: bug: EventBridge event-pattern not correctly matched on detail content
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

The provided example does not put a message in SQS.

### Expected Behavior

The provided example puts a message in SQS.

### How are you starting LocalStack?

With a docker-compose file

### Steps To Reproduce

The following scripts returns no items in SQS, but it should. The bit detail: foo is not matched.

```
awslocal events create-event-bus \
    --name detail-bus-1

awslocal sqs create-queue \
    --queue-name "detail-sqs-1"

awslocal events put-rule \
    --name "detail-rule-1" \
    --event-bus-name "detail-bus-1" \
    --event-pattern '{ "detail": { "foo": [ { "exists": true } ] } }'

awslocal events put-targets \
    --rule "detail-rule-1" \
    --event-bus-name "detail-bus-1" \
    --targets "Id"="detail-target-rule-1","Arn"="arn:aws:sqs:us-east-1:000000000000:detail-sqs-1","InputPath"="$.detail"

awslocal events put-events --entries '[{"DetailType": "test", "Source": "aws.test", "Detail": "{ \"foo\": \"bar\" }", "EventBusName": "detail-bus-1"}]'
awslocal sqs receive-message --queue-url http://localhost:4566/000000000000/detail-sqs-1
```

### Environment

```markdown
- OS: macOS 13.3.1
- LocalStack: 1.4.0
```

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by maxhoheiser**
## Motivation
As reported in #8512 and #7305 the current behavior does not fully match the aws cli behavior regarding the EventBridge pattern `{"exists": True}`.

This rule:
```
{
	"source": ["core.update-account-command"],
	"detail-type": ["core.update-account-command"],
	"detail": {"key": [{"exists": True}]},
}
```

in combination with this message:
```
{
	"version": "0",
	"id": "<uuid:2>",
	"detail-type": "core.update-account-command",
	"source": "core.update-account-command",
	"account": "111111111111",
	"time": "date",
	"region": "<region>",
	"resources": [],
	"detail": {
	    "key":"value",
		"payload":"baz"
	}
}
```

does not match on LocalStack, but does correctly match on AWS.

The issue arises from the prefix `exists` not being considered.

Furthermore, the opposite pattern `[("exists"): False]` was also not correctly matched

This rule:
```
{
	"source": ["core.update-account-command"],
	"detail-type": ["core.update-account-command"],
	"detail": {"key": [{"exists": False}]},
}
```

in combination with this message:
```
{
	"version": "0",
	"id": "<uuid:2>",
	"detail-type": "core.update-account-command",
	"source": "core.update-account-command",
	"account": "111111111111",
	"time": "date",
	"region": "<region>",
	"resources": [],
	"detail": {
	    "no-key":"no-value",
		"payload":"baz"
	}
}
```

## Changes
Adding a condition to check for the prefix `exists` and use the boolean value in combination with the condition if there value to return exists for the pattern `[("exists"): True]`.

The pattern `[("exists"): False]` needs different handling, since per definition, the getter of the message event for the in the pattern defined key will always fail to the fallback object. To still send the message correctly, an additional check is put in place handling this particular instance when the return value is not presented, that is only `True` in the condition that the pattern prefix equal `"exists"` see (`provider.py` line 516).

Fixes #8512 and fixes #7305 

## Testing

Polling for sqs messages was changed to long poll due to inconsistencies when generating the snapshots against aws with messages not being received.

### CODE REVIEW

**Review by dominikschubert — APPROVED** [review]
PR is already in a good state to merge, but I have a few suggestions. They can be either tackled in this PR, later or just added as TODOs for future us :+1: 

Sorry you had to deal with such an annoying test as one of your first bigger AWS tests :sweat_smile:  Glad we could find a reliable solution in the end.

File: localstack/services/events/provider.py [hunk 1]
```diff
@@ -510,6 +513,14 @@ def get_two_lists_intersection(lst1: List, lst2: List) -> List:
     return lst3
 
 
+def event_pattern_prefix_bool_filter(event_pattern_filter_value_list: List[Dict[str, Any]]) -> bool:
```
[the remark this answers is not in the document]
Comment (maxhoheiser) [reply localstack/services/events/provider.py]: resolved in : [18299c1](https://github.com/localstack/localstack/pull/9931/commits/18299c122d0239fb39bf6847d00b6f60108738e9)

File: tests/aws/services/events/test_events.py [hunk 2]
```diff
@@ -497,6 +503,64 @@ def test_put_events_with_rule_anything_but_to_sqs(
         )
         snapshot.match("rule-anything-but", messages)
 
+    @markers.aws.validated
+    def test_put_events_with_rule_exists_to_sqs(self, put_events_with_filter_to_sqs, snapshot):
+        snapshot.add_transformer(
```
[the remark this answers is not in the document]
Comment (maxhoheiser) [reply tests/aws/services/events/test_events.py]: fixed in: [e3283a4](https://github.com/localstack/localstack/pull/9931/commits/e3283a42126497687453f95af5a1e2536a56a63e)

File: tests/aws/services/events/test_events.py [hunk 3]
```diff
@@ -497,6 +503,64 @@ def test_put_events_with_rule_anything_but_to_sqs(
         )
         snapshot.match("rule-anything-but", messages)
 
+    @markers.aws.validated
+    def test_put_events_with_rule_exists_to_sqs(self, put_events_with_filter_to_sqs, snapshot):
```
[the remark this answers is not in the document]
Comment (maxhoheiser) [reply tests/aws/services/events/test_events.py]: fixed in: [e3283a4](https://github.com/localstack/localstack/pull/9931/commits/e3283a42126497687453f95af5a1e2536a56a63e)

**Pull request merged**